projects/plugins/jetpack/modules/memberships/class-jetpack-memberships.php

```diff
@@ -203,6 +203,17 @@ public function init_hook_action() {
 		add_filter( 'rest_api_allowed_post_types', array( $this, 'allow_rest_api_types' ) );
 		add_filter( 'jetpack_sync_post_meta_whitelist', array( $this, 'allow_sync_post_meta' ) );
 		$this->setup_cpts();
+
+		if ( Jetpack::is_module_active( 'subscriptions' ) && jetpack_is_frontend() ) {
+			add_action( 'wp_logout', array( $this, 'subscriber_logout' ) );
```

Comment:
To be more clear, we should clear the cookie on WPCOM when login in, not login out.

Reply:
I want it to be called every time the user clicks:
1. The "Log out" link provided by the Subscriber Login block
2. The "Switch accounts" link provided by the Paywall block

Basically, we want to clear the premium content cookie on every logout action as long as the subscription module is active. 

What is the use case for clearing it out on log-in?

Reply:
> What is the use case for clearing it out on log-in?

Simple sites don't use the cookie when user is logged in.